src/platform/unittest/DataPathTest.cpp

```diff
@@ -179,15 +183,65 @@ struct DataPathTest : public ::testing::TestWithParam<int32_t>
         }
     }
 
+    //
+    // Helper to return a new unspecified IPv4 address and port to use.
+    //
+    QuicAddr
+    GetNewUnspecIPv4(bool randomPort = true)
+    {
+        QuicAddr ipv4Copy = UnspecIPv4;
+        if (randomPort) { ipv4Copy.SockAddr.Ipv4.sin_port = GetNextPort(); }
+        else { ipv4Copy.SockAddr.Ipv4.sin_port = 0; }
+        return ipv4Copy;
+    }
+
+    //
+    // Helper to return a new unspecified IPv4 address and port to use.
+    //
+    QuicAddr
+    GetNewUnspecIPv6(bool randomPort = true)
+    {
+        QuicAddr ipv6Copy = UnspecIPv6;
+        if (randomPort) { ipv6Copy.SockAddr.Ipv6.sin6_port = GetNextPort(); }
+        else { ipv6Copy.SockAddr.Ipv6.sin6_port = 0; }
+        return ipv6Copy;
+    }
+
+    //
+    // Helper to return a new unspecified IPv4 or IPv6 address based on the test data.
+    //
+    QuicAddr
+    GetNewUnspecAddr(bool randomPort = true)
+    {
+        int addressFamily = GetParam();
+
+        if (addressFamily == 4) {
+            return GetNewUnspecIPv4(randomPort);
+        } else if (addressFamily == 6) {
+            return GetNewUnspecIPv6(randomPort);
+        } else {
+            GTEST_NONFATAL_FAILURE_("Malconfigured test data; This should never happen!!");
+            return QuicAddr();
+        }
+    }
+
     static void SetUpTestSuite()
     {
         //
         // Initialize a semi-random base port number.
         //
         NextPort = 50000 + (CxPlatCurThreadID() % 10000) + (rand() % 5000);
 
-        LocalIPv4.Resolve(QUIC_ADDRESS_FAMILY_INET, "localhost");
-        LocalIPv6.Resolve(QUIC_ADDRESS_FAMILY_INET6, "localhost");
+        if (UseDuoNic) {
+            LocalIPv4.Resolve(QUIC_ADDRESS_FAMILY_INET, "192.168.1.11");
+            LocalIPv6.Resolve(QUIC_ADDRESS_FAMILY_INET6, "fc00::1:11");
+        } else {
+            LocalIPv4.Resolve(QUIC_ADDRESS_FAMILY_INET, "localhost");
+            LocalIPv6.Resolve(QUIC_ADDRESS_FAMILY_INET6, "localhost");
```

Comment:
I _think_ this will still not work on Linux, but I'm not sure. Because it uses a different name for v6 localhost (look in msquic_posix.h I think).

Reply:
The linux test passed, but I'll still switch this to use QUIC_LOCALHOST_FOR_AF.